src/ast/passes/codegen_llvm.cpp

```diff
@@ -2635,6 +2634,35 @@ void CodegenLLVM::visit(AttachPoint &)
   // Empty
 }
 
+// For older 5.2 kernels, we need to bound CPU ID by MAX_CPU_ID to pass
+// BPF verifier on older kernels.
+//
+// However, for programs that do many scratch buffer lookups, this causes
+// many jumps which means a complex program may be rejected by the BPF
+// verifier for >= 8192 jumps (current limit).
+//
+// To minimize jumps, we bound the CPU ID at the beginning of each function
+// and cache the value in a register. This ensures we only add one
+// additional jump per function. Note we cannot cache this value in memory
+// since memory is shared between all CPUs
```

Comment:
> cache this value in memory

What do you mean by memory?

Reply:
To be clear, I mean we can't call `CreateStore()` and save it to some location in memory. Because memory of the BPF script is shared between all CPUs (at least AFAIK), so we can't have CPU A write the bounded CPU ID and CPU B load the bounded CPU ID because it'd yield the wrong result.

Basically saying we have to keep it in a register

Reply:
I see. Right, unless it was a percpu array. In which case it's recursive. Maybe replace "memory" with "global variable"? I think that would make it less confusing

Reply:
Fixed in latest version!